File: src/my_project_typescript/medium_problems_typescript/from51to100/twoSum.ts
function twoSum(nums: number[], target: number): number[] {

    let tempObj: {[key:number]: number} = {}
    let lenNums: number = nums.length

    for (let i = 0; i < lenNums; i++){
        if (target-nums[i] in tempObj){
            return [tempObj[target-nums[i]],i]
        }
        tempObj[nums[i]] = i
    }
    


};

File: src/my_project/interviews/top_150_questions_round_18/two_sum.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        dic_answer = dict()

        for k, v in enumerate(nums):

            if v not in dic_answer:
                dic_answer[target - v] = k 
            else: 
                return [dic_answer[v], k]
            
        return -1
